### PR TITLE
Update unoptimized images policy

### PR DESCRIPTION
**Pull request by loonybear**
Added a section for Origin Trials
Updated new syntax
Added suggested threshold for features.

### CODE REVIEW

**Comment by loonybear** [comment]
@jpmedley @clelland please take a look. Thanks

File: policies/optimized-images.md [hunk 1]
```diff
@@ -14,19 +14,42 @@ Optimized image policies are aiming to solve problems with sites shipping images
...
+    *   Lossless images used in rendering must not include too much metadata.
+    *   Lossless images should be in one of the modern image formats that yield large byte savings and performance improvement.
+    *   A lossless `<img>` element should not exceed a byte-per-pixel ratio of _***X***_, with a fixed _**1KB**_ overhead allowance. For a W x H image, the file size threshold is calculated as `W x H x X + 1024` (where X is specified in the policy). Any image whose file size exeeds the limit will be blocked.    
+
+**Note**: We want to allow developers the ability to make the final decision about the tradeoffs they make. _***X***_ means developers can specify the "value" of the policy. For example, `oversized-images *(2)` specifies the maximum ratio, 2, images are allowed to oversize by.
+
+
+## Experiment image policies with Origin Trials
+
+Image policieis are shipped in Chrome M75 via Origin Trials.
```
Comment (clelland) [inline policies/optimized-images.md]: Sp: "policies"

File: policies/optimized-images.md [hunk 2]
```diff
@@ -14,19 +14,42 @@ Optimized image policies are aiming to solve problems with sites shipping images
...
+*   **["unoptimized-lossless-images" policy](#unoptimized-{lossy,lossless}-images)**
+    *   Lossless images used in rendering must not include too much metadata.
+    *   Lossless images should be in one of the modern image formats that yield large byte savings and performance improvement.
+    *   A lossless `<img>` element should not exceed a byte-per-pixel ratio of _***X***_, with a fixed _**1KB**_ overhead allowance. For a W x H image, the file size threshold is calculated as `W x H x X + 1024` (where X is specified in the policy). Any image whose file size exeeds the limit will be blocked.    
+
+**Note**: We want to allow developers the ability to make the final decision about the tradeoffs they make. _***X***_ means developers can specify the "value" of the policy. For example, `oversized-images *(2)` specifies the maximum ratio, 2, images are allowed to oversize by.
+
+
+## Experiment image policies with Origin Trials
```
Comment (clelland) [inline policies/optimized-images.md]: Can you mention in the heading that this particular section is Chrome-specific?

File: policies/optimized-images.md [hunk 3]
```diff
@@ -1,8 +1,8 @@
 #  Image Policies Explainer
 
-loonybear@, last updated: 10/30/2018
+loonybear@, last updated: 03/27/2019
 
-<span style="color:#38761d;">Status: Draft Proposal</span>
+<span style="color:#38761d;">Status: in [Origin Trials](https://github.com/GoogleChrome/OriginTrials) in M75</span>
```
Comment (clelland) [inline policies/optimized-images.md]: This is still a proposal, but in Chrome is part of an origin trial... not sure how to make that clear here, but we should make the distinction.

File: policies/optimized-images.md [hunk 4]
```diff
@@ -14,19 +14,42 @@ Optimized image policies are aiming to solve problems with sites shipping images
...
+*   **["unoptimized-lossy-images" policy](#unoptimized-{lossy,lossless}-images)**
+    *   Lossy images used in rendering must not include too much metadata.
+    *   Lossy images should be in one of the modern image formats that yield large byte savings and performance improvement.
+    *   A lossy `<img>` element should not exceed a byte-per-pixel ratio of _***X***_, with a fixed _**1KB**_ overhead allowance. For a W x H image, the file size threshold is calculated as `W x H x X + 1024` (where X is specified in the policy). Any image whose file size exeeds the limit will be blocked.
+*   **["unoptimized-lossless-images" policy](#unoptimized-{lossy,lossless}-images)**
+    *   Lossless images used in rendering must not include too much metadata.
+    *   Lossless images should be in one of the modern image formats that yield large byte savings and performance improvement.
+    *   A lossless `<img>` element should not exceed a byte-per-pixel ratio of _***X***_, with a fixed _**1KB**_ overhead allowance. For a W x H image, the file size threshold is calculated as `W x H x X + 1024` (where X is specified in the policy). Any image whose file size exeeds the limit will be blocked.
```
Comment (clelland) [inline policies/optimized-images.md]: Maybe make a note here, or in the origin trial section, that we are experimenting with an alternate number of bytes for overhead?

Also, I think that the overhead allowed for `unoptimized-lossless-images` is 10k currently, and 1k is for `unoptimized-lossless-images-strict`, but check with @paulmeyer90, or take a look in the code.

File: policies/optimized-images.md [hunk 5]
```diff
@@ -14,19 +14,42 @@ Optimized image policies are aiming to solve problems with sites shipping images
...
+
+Image policieis are shipped in Chrome M75 via Origin Trials.
+
+Request a token to try the origin trial on your own origin:
+   * Provide the token on any pages in your origin using an `Origin-Trial` HTTP header:
+   ```
+   Origin-Trial: **token as provided in the developer console**
+   ```
+
+   * Specify an image policy via HTTP header `Feature-Poliy` header (see below for more details).
```
Comment (clelland) [inline policies/optimized-images.md]: Sp: 'Policy"

File: policies/optimized-images.md [hunk 6]
```diff
@@ -39,60 +62,70 @@ Optimized image policies introduce a set of restrictions (policies) on images th
 
 On a web page, the number of pixels of a container determines the resolution of an image served inside. It is unnecessary to use an image that is much larger than what the viewing device can actually render; for example, serving a desktop image to mobile contexts, or serving an image intended for high-pixel-density screens to a low-pixel-density device. This results in unnecessary network traffic and downloaded bytes. `oversized-images` is a policy controlled feature that restricts images to be no more than X times bigger than the container size.
 
-When a document is disallowed to use `oversized-images` policy, its `<img>` elements that are more than X times larger than its container size will be rendered as a placeholder image.
-
+When a document disallows the `oversized-images` policy, the `<img>` elements that are more than X times larger than the container size will be replaced with placeholder images.
 
-#### Specification
+To try the `oversized-images` policy, register a token [here](https://developers.chrome.com/origintrials/#/trials/active) and specify the policy via HTTP `Feature-Policy` header (see section above for more details).
```
Comment (clelland) [inline policies/optimized-images.md]: Mention that this is Chrome-specific, or move this out of the policy discussion

File: policies/optimized-images.md [hunk 7]
```diff
@@ -39,60 +62,70 @@ Optimized image policies introduce a set of restrictions (policies) on images th
...
+all `<img>` elements will be allowed and rendered correctly by default.
 
-    ```html
-    <iframe allow="oversized-images(4)"></iframe>
-    ```
-    That would apply a policy in which the maximum oversizing ratio allowed is set to 4.
+- An `oversized-images` policy can be specified via:
 
-    Feature policies combine in subframes, and the minimum value of the downscaling ratio will be applied, so if that frame embedded another, which the syntax:
+    **1. [HTTP `Feature-Policy`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Feature-Policy) response header:**
```
Comment (clelland) [inline policies/optimized-images.md]: You could link to the spec here, too -- https://w3c.github.io/webappsec-feature-policy/#feature-policy-http-header-field

File: policies/optimized-images.md [hunk 8]
```diff
@@ -140,71 +173,74 @@ For an `<img>` element, if neither the width or the height of the source image e
...
     ```html
-    <iframe allow="unoptimized-images(2048, 0.2)"></iframe>
+    <iframe src="https://example.com" allow="unoptimized-lossless-images 'self'(0.8) https://foo.com(1);">
     ```
-    then the child frame would be allowed to render images with metadata size limit of 1500 bytes and byte-per-pixel ratio of 0.2. 
-
-- The default allowlist for `unoptimized-images` is `*`. This means for pages of all origins, `<img>` elements whose file sizes exceeds the compression ratio will be allowed and rendered correctly.
 
+    In this example, `unoptimized-lossless-images` is **disabled everywhere except on the origin of the main document and on `https://foo.com`**. On the origin of the main document, any non JPEG `<img>` element whose file size exeeds the maximum file size allowance (with pite-per-pixel ratio set to 0.8) will be replaced with a placeholder image. On 'https://foo.com', any non JPEG `<img>` element whose file size exeeds the maximum file size allowance (with pite-per-pixel ratio set to 1) will be replaced with a placeholder image. **`<img>` elements on any other origins whose file size exeeds 1KB will be replaced with placeholder images**.
```
Comment (clelland) [inline policies/optimized-images.md]: sp: "byte-per-pixel" (twice)

File: policies/optimized-images.md [hunk 9]
```diff
@@ -140,71 +173,74 @@ For an `<img>` element, if neither the width or the height of the source image e
...
 
-- A `unoptimized-images` policy can be specified via:
+- Feature policies combine in subframes, and the minimum value of the bite-per-pixel ratio will be applied, so if a frame, whose maximum bite-per-pixel ratio is set to 0.9 for unoptimized-lossy-images, embedded another, which the syntax:
```
Comment (clelland) [inline policies/optimized-images.md]: sp: "byte-per-pixel" (twice)

File: policies/optimized-images.md [hunk 10]
```diff
@@ -39,60 +62,70 @@ Optimized image policies introduce a set of restrictions (policies) on images th
 
 On a web page, the number of pixels of a container determines the resolution of an image served inside. It is unnecessary to use an image that is much larger than what the viewing device can actually render; for example, serving a desktop image to mobile contexts, or serving an image intended for high-pixel-density screens to a low-pixel-density device. This results in unnecessary network traffic and downloaded bytes. `oversized-images` is a policy controlled feature that restricts images to be no more than X times bigger than the container size.
 
-When a document is disallowed to use `oversized-images` policy, its `<img>` elements that are more than X times larger than its container size will be rendered as a placeholder image.
-
+When a document disallows the `oversized-images` policy, the `<img>` elements that are more than X times larger than the container size will be replaced with placeholder images.
```
Comment (clelland) [inline policies/optimized-images.md]: We should probably say "more than X times larger than the container size in either dimension" to make it clear that we're looking at both width and height separately, and not just counting pixels.

**Comment by clelland** [comment]
Merged, thanks!

**Pull request merged**